src/NHibernate/Loader/Loader.cs

```diff
@@ -980,22 +981,63 @@ private void CheckVersion(int i, IEntityPersister persister, object id, object e
 			for (int i = 0; i < cols; i++)
 			{
 				object obj = null;
+				ILoadable persister = persisters[i];
+
 				EntityKey key = keys[i];
 
 				if (key == null)
 				{
-					// do nothing
-					/* TODO NH-1001 : if (persisters[i]...EntityType) is an OneToMany or a ManyToOne and
-					 * the keys.length > 1 and the relation IsIgnoreNotFound probably we are in presence of
-					 * an load with "outer join" the relation can be considerer loaded even if the key is null (mean not found)
+					if (cols == 1 || CollectionOwners == null)
+					{
+						continue;
+					}
+
+					int position = GetPosition(i);
```

Comment:
I suspect that `position` always equals to `i+1`.

Reply:
Ok, except QueryOver trick with skip...

Reply:
My main point was to provide example when order of entities in result set is different from mapping (so mapping based index check is wrong).  IMHO number of queries are not relevant - but I can double check later if you think it's useful (just let me know).
Theoretically this fix should save 1 additional request for checking Department2 but all other `not-found` ignore properties should still trigger requests.
So I think 4 statements in total : 1 for main query + 3 for Department1, Department3, Address "not found" ignore check.

And regarding current fix. This Aliases parse logic doesn't feel reliable and looks like depends on some internal knowledge of how aliases are generated. I would avoid such dependencies.